File: 2315-count-asterisks/2315-count-asterisks.java
class Solution {

    public int countAsterisks(String s) {
        String[] spl = s.split("\\|");
        int result = 0;
        int i = 0;
        while (i < spl.length) {
            int c = 0;
            String str = spl[i];
            for (int j = 0; j < str.length(); j++) {
                char a = str.charAt(j);
                if (a == '*') {
                    c++;
                }
            }
            result += c;
            i += 2;
        }
        return result;
    }
}

File: 2315-count-asterisks/2315-count-asterisks.cs
public class Solution
{

  private int CountChars(string s, char c)
  {
    int result = 0;
    foreach (char item in s)
    {
      if (item == c)
      {
        result++;
      }
    }

    return result;
  }

  public int CountAsterisks(string s)
  {
    int result = 0;
    var spl = s.Split("|");
    int i = 0;
    while (i < spl.Length)
    {
      string word = spl[i];
      result += CountChars(word, '*');

      i += 2;
    }
    return result;
  }
}
